### PR TITLE
Relocate optimization specific processes to lava-optimization and out of lava/proc

### PR DESCRIPTION
**Pull request by GaboFGuerra**
Objective of pull request: Relocate optimization specific processes to lava-optimization and out of lava/proc.

## Pull request checklist

Your PR fulfills the following requirements:
- [x] Tests are part of the PR (for bug fixes / features)
- [ ] [Docs](https://github.com/lava-nc/docs) reviewed and added / updated if needed (for bug fixes / features)
- [x] PR conforms to [Coding Conventions](https://lava-nc.org/developer_guide.html#coding-conventions)
- [x] [PR applys BSD 3-clause or LGPL2.1+ Licenses](https://lava-nc.org/developer_guide.html#add-a-license) to all code files
- [x] Lint (`pyb`) passes locally
- [ ] Build tests (`pyb -E unit`) or (`python -m unittest`) passes locally

## Pull request type

 

 

Please check your PR type:
- [x] Refactoring 

## What is the current behavior?

- Optimization specific processes are exposed under lava/proc which is intended for generic processes.

## What is the new behavior?

- Processes specific to the optimization library have been moved under lava/lib/optimization/solvers/generic/processes.

## Does this introduce a breaking change?

- [x] Yes

## Supplemental information

The import path for ReadGate, QuboScif and CostIntegrator are changed, they are not longer under lava.proc but under lava.optimization.solvers.generic.processes

### CODE REVIEW

**Comment by AlessandroPierro** [comment]
This PR shouldn't contain the generator?MIS utils, since they are also included (in their updated version) in #71 .

**Review by calixphd — APPROVED** [review]

**Comment by AlessandroPierro** [comment]
Tests on Scif are failing with:

`lava.magma.compiler.exceptions.NoProcessModelFound: No ProcessModels found that implement Process '<lava.lib.optimization.solvers.generic.scif.process.QuboScif object at 0x7ffa90c57d60>'`

**Comment by phstratmann** [comment]
> Tests on Scif are failing with:
> 
> `lava.magma.compiler.exceptions.NoProcessModelFound: No ProcessModels found that implement Process '<lava.lib.optimization.solvers.generic.scif.process.QuboScif object at 0x7ffa90c57d60>'`

And with
image

File: tests/lava/lib/optimization/solvers/generic/scif/test_models.py [hunk 1]
```diff
@@ -0,0 +1,428 @@
+
```
Comment (phstratmann) [inline tests/lava/lib/optimization/solvers/generic/scif/test_models.py:1]: Wrong license header for public repo

File: tests/lava/lib/optimization/solvers/generic/scif/test_process.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+
```
Comment (phstratmann) [inline tests/lava/lib/optimization/solvers/generic/scif/test_process.py:1]: Wrong license header for public repo.

**Review by phstratmann — APPROVED** [review]
Before merging, please ensure that the license headers are correct! Currently, they are wrong for the test files, see my comments.

Didn't review the code in detail, as it has been reviewed previously. Only the new places for the files.

Thanks for your great work :-)

**Comment by GaboFGuerra** [comment]
> 

solved

**Pull request merged**